client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -383,11 +383,7 @@ public void write(PendingEvent event) {
                                            event.getExpectedOffset()));
             } catch (ConnectionFailedException e) {
                 log.warn("Connection " + writerId + " failed due to: ", e);
-                try {
-                    getConnection(); // As the messages is inflight, this will perform the retransmition.
-                } catch (SegmentSealedException e2) {
-                    return;
-                }
+                reconnect(); // As the messages is inflight, this will perform the retransmition.
```

Comment:
Typos, it should be `As the message is inflight, this will perform the retransmission.`